> I don't know if mypy can fail if we pass a Defered instance.

It does fail.  Consider:

```
from twisted.internet.defer import Deferred
d = Deferred[int]()
d.callback(Deferred[int]())
```
produces:

```
src/twisted/x.py:3:12: error: Argument 1 to "callback" of "Deferred" has incompatible type "Deferred[int]"; expected "int | Failure"  [arg-type]
src/twisted/x.py:3:12: note: Maybe you forgot to use "await"?
Found 1 error in 1 file (checked 1 source file)
```

--- a/src/twisted/newsfragments/12122.feature
+++ b/src/twisted/newsfragments/12122.feature
@@ -1,1 +1,1 @@
-``twisted.internet.defer.Deferred`` operations use less CPU.
+twisted.internet.defer.Deferred.callback() and twisted.internet.defer.Deferred.addCallbacks() no longer use `assert` to check the type of the arguments. You should now use type checking to validate your code. These changes were done to reduce the CPU usage.

